src/mesh_analysis/tracers/waku_tracer.py

```diff
@@ -0,0 +1,37 @@
+# Python Imports
+import re
+import pandas as pd
+from typing import List
+
+
+# Project Imports
+from src.mesh_analysis.tracers.message_tracer import MessageTracer
+
+
+class WakuTracer(MessageTracer):
+
+    def __init__(self):
+        # TODO: Improve patterns as:
+        # - Different patterns (received, sent, dropped)
+        # - Once one pattern search is completed, stop search for it in the logs (ie: Announce Address)
+        self._patterns = [re.compile(
+            r'^(\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}\.\d+\+\d{2}:\d{2}) .* msg_hash=([a-fA-F0-9x]+) .* sender_peer_id=([A-Za-z0-9]+)$'),
+            re.compile(r'.* Announcing addresses .*\[([^]]+)\]$')]
+
+    def trace(self, parsed_logs: List) -> pd.DataFrame:
+        df = self._trace_message_in_logs(parsed_logs)
+
+        return df
+
+    def _trace_message_in_logs(self, parsed_logs: List) -> pd.DataFrame:
+        parsed_logs = [log for log in parsed_logs if len(log[0]) > 0]
```

Comment:
Not that it matters a lot in this case, but as a general rule I'd recommend using generators over list comprehensions when possible.